src/werkzeug/formparser.py

```diff
@@ -412,7 +479,20 @@ def validate_boundary(self, boundary):
             # the assert is skipped.
             self.fail("Boundary longer than buffer size")
 
-    def parse_lines(self, file, boundary, content_length, cap_at_buffer=True):
+    def parse_lines(
```

Comment:
I'm a bit confused by 
```python
iterator = chain(
  make_line_iter(
    file,
    limit=content_length,
    buffer_size=self.buffer_size,
    cap_at_buffer=cap_at_buffer,
  ),
  _empty_string_iter,
)
```
`make_line_iter` returns an iterator of `bytes` but `_empty_string_iter` returns a iterable of (empty) strings. Therefore `iterator` should have the type `Union[str, bytes]` which means that the function should return parts of ``('cont', str or bytes)`` rather than ``('cont', str)`` but this seems odd. Is that the expected behaviour?

Reply:
This is probably an artifact of adding then removing Python 2/3 compatibility. This should probably always return bytes, since that's what's coming in from the file.

Reply:
Yeah, that's what confused me – should I update `_empty_string_iter` to return `b""` (bytes) instead of `""` (strings)?

Reply:
Yes, but let's keep any fixes to the code as a separate commit at the end, so it's easy to review the types vs the fixes.

Reply:
Sounds good! My plan is to split things out into a large number of small commits so that it's easier to review/revert changes.